### PR TITLE
AlertGroups: Generate models.gen.ts from models.cue

### PR DESCRIPTION
**Pull request by Clarity-89**
**What is this feature?**

Add `models.cue` for `AlertGroups` panel and use the types generated from it. 

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/48761

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/alertGroups/models.cue [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	"github.com/grafana/thema"
+)
+
+Panel: thema.#Lineage & {
```
[the remark this answers is not in the document]
Comment (sdboyer) [reply public/app/plugins/panel/alertGroups/models.cue:21]: appreciate it, ty

File: public/app/plugins/panel/alertGroups/composable_panelcfg.cue [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Grafana Labs
```
Comment (joanlopez) [inline public/app/plugins/panel/alertGroups/composable_panelcfg.cue:1]: Based on new convention, starting from [here](https://github.com/grafana/grafana/pull/61928/files), I think this file should be named just `panelcfg.cue`.

@sdboyer could we somehow force this kind of naming convention? (e.g. making the jenny -gen.go- to only read those; and/or display a warning)

Comment (Clarity-89) [reply public/app/plugins/panel/alertGroups/composable_panelcfg.cue]: good catch! Should be fixed now.

Comment (joanlopez) [reply public/app/plugins/panel/alertGroups/composable_panelcfg.cue]: Excellent!

**Review by sdboyer — APPROVED** [review]

**Pull request merged**